File: test/interface/golden/sum/NewType.ml
type newType =
  | NewType of int

let encodeNewType x =
  match x with
  | NewType y0 ->
     Aeson.Encode.int y0

let decodeNewType json =
  match Aeson.Decode.int json with
  | v -> Js_result.Ok (NewType v)
  | exception Aeson.Decode.DecodeError msg -> Js_result.Error ("decodeNewType: " ^ msg)

File: test/interface/golden/lib/js/sum/NewType.js
'use strict';

var Block        = require("bs-platform/lib/js/block.js");
var Js_exn       = require("bs-platform/lib/js/js_exn.js");
var Aeson_decode = require("bs-aeson/lib/js/src/Aeson_decode.js");

function encodeNewType(x) {
  return x[0];
}

function decodeNewType(json) {
  var exit = 0;
  var v;
  try {
    v = Aeson_decode.$$int(json);
    exit = 1;
  }
  catch (raw_exn){
    var exn = Js_exn.internalToOCamlException(raw_exn);
    if (exn[0] === Aeson_decode.DecodeError) {
      return /* Error */Block.__(1, ["decodeNewType: " + exn[1]]);
    } else {
      throw exn;
    }
  }
  if (exit === 1) {
    return /* Ok */Block.__(0, [/* NewType */[v]]);
  }
  
}

exports.encodeNewType = encodeNewType;
exports.decodeNewType = decodeNewType;
/* No side effect */
